solidity/dashboard/src/components/Modal.jsx

```diff
@@ -72,10 +79,85 @@ export const ModalContext = React.createContext({
+  const dispatch = useDispatch()
   const [modalComponent, setModalComponent] = useState()
   const [isOpen, setIsOpen] = useState(false)
   const [modalOptions, setModalOptions] = useState(null)
   const awaitingPromiseRef = useRef()
+  const modal = useSelector((state) => state.modal)
+
+  useEffect(() => {
+    if (modal.isOpen) {
+      switch (modal.emittedEvent) {
+        case EVENTS.COVERAGE_POOLS.WITHDRAWAL_COMPLETED:
+          openModal(<ClaimTokensModal transactionFinished={true} amount={modal.additionalData?.amount} onCancel={closeModal}/>, {
+            closeModal: closeModal,
+            title: "Claim tokens",
+            classes: {
+              modalWrapperClassName: "modal-wrapper__claim-tokens",
+            },
+          })
+          break;
+        case EVENTS.COVERAGE_POOLS.WITHDRAWAL_INITIATED:
+          openModal(<InitiateCovPoolsWithdrawModal
+            transactionFinished={true}
+            amount={modal.additionalData?.covAmount}
+            onCancel={closeModal}
+          />, {
+            closeModal: closeModal,
+            title: "Withdraw",
+            classes: {
+              modalWrapperClassName: "modal-wrapper__initiate-withdrawal",
+            },
+          })
+          break;
+        case EVENTS.COVERAGE_POOLS.RE_WITHDRAWAL_INITIATED:
+          openModal(<InitiateCovPoolsWithdrawModal
+            transactionFinished={true}
+            amount={modal.additionalData?.covAmount}
+            onCancel={closeModal}
+          />, {
+            closeModal: closeModal,
+            title: "Re-initiate withdrawal",
+            classes: {
+              modalWrapperClassName: "modal-wrapper__initiate-withdrawal",
+            },
+          })
+          break;
+        case EVENTS.COVERAGE_POOLS.ADD_BALANCE_TO_WITHDRAWAL:
+          openModal(<IncreaseWithdrawalModal
+            transactionFinished={true}
+            pendingWithdrawalBalance={modal.additionalData?.covAmount}
+            addedAmount={"0"}
+            onCancel={closeModal}
+          />, {
+            closeModal: closeModal,
+            title: "Re-initiate withdrawal",
+            classes: {
+              modalWrapperClassName: "modal-wrapper__initiate-withdrawal",
+            },
+          })
+          break;
+        case EVENTS.COVERAGE_POOLS.COV_TOKEN_TRANSFERED_TO_USER:
+          openModal(<InitiateDepositModal
+            transactionFinished={true}
+            amount={modal.additionalData?.amount}
+            balanceAmount={modal.additionalData?.updatedCovBalance}
+            estimatedBalanceAmountInKeep={modal.additionalData?.estimatedKeepBalance}
+            onCancel={closeModal}
+          />, {
+            closeModal: closeModal,
+            title: "Deposit",
+            classes: {
+              modalWrapperClassName: "modal-wrapper__initiate-withdrawal",
+            },
+          })
+          break;
+        default:
+          break;
+      }
+    }
```

Comment:
Can we use the same approach as we use for displaying [messages](https://github.com/keep-network/keep-core/blob/main/solidity/dashboard/src/components/Message.jsx#L18-L28)?

Reply:
Fixed in cd9abad98ab755f57b1cd5ff39a2afce4545fe20